src/main/java/org/apache/sling/api/resource/observation/ResourceChangeListener.java

```diff
@@ -92,11 +92,41 @@
      * If this property is missing, added, removed and changed events for resources
      * and added and removed events for resource providers are reported.
      * If this property is invalid, the listener is ignored. The type of the property
-     * must either be String, or a String array. Valid values are the constants from
-     * {@link ResourceChange.ChangeType}.
+     * must either be String, or a String array. Valid values are the constants from this class
+     * whose names are starting with {@code CHANGE_}. They map to one of the values of {@link ResourceChange.ChangeType}.
      */
     String CHANGES = "resource.change.types";
 
+    /**
+     * String constant for {@link ResourceChange.ChangeType#ADDED}.
+     * @since 1.3.0 (Sling API Bundle 2.23.0)
+     */
+    String CHANGE_ADDED = "ADDED";
```

Comment:
Any reason for not directly using `ResourceChange.ChangeType.ADDED.toString()`? Same for the other constants.

Reply:
Yes, that is not considered a compile time constant, compare with the description of https://issues.apache.org/jira/browse/SLING-9555 and e.g. https://stackoverflow.com/a/13253879.